mkosi/config.py

```diff
@@ -5134,7 +5134,7 @@ def parse_config(
                 history.pop(s.dest, None)
                 continue
 
-            if s.dest in context.cli and context.cli[s.dest] != history[s.dest]:
+            if s.dest in context.cli and s.dest in history and context.cli[s.dest] != history[s.dest]:
```

Comment:
Even if it is not in the history it still gets ignored, right? So the warning should still be omitted in that case?

```suggestion
            if s.dest in context.cli and not (s.dest in history and context.cli[s.dest] == history[s.dest]):
```

or

```suggestion
            if s.dest in context.cli and context.cli[s.dest] != history.get(s.dest, context.cli[s.dest]):
```